File: farmacia.js
let inventario = {};
let tVentas= 0;

function basededatos (){

    let nombre = prompt("Ingrese el nombre del medicamento");

    let precio = parseFloat(prompt("Ingrese el precio del medicamento"));

    let cantidad = parseInt(prompt("Ingrese la cantidad disponible"));

   inventario[nombre] = { cantidad: cantidad, precio: precio };
    console.log("Medicamento registrado");
}

function consultar() {
    console.log("INVENTARIO:");
    for (let med in inventario) {
        console.log(
            med +
            " - Cantidad: " +
            inventario[med].cantidad +
            " - Precio: " +
            inventario[med].precio
        );
    }
}

function vender() {
    let nombre = prompt("Medicamento a vender:");

    if (inventario[nombre]) {
        let cant = Number(prompt("Cantidad a vender:"));

        if (cant <= inventario[nombre].cantidad) {
            inventario[nombre].cantidad -= cant;
            totalVentas += cant * inventario[nombre].precio;
            console.log("Venta realizada");
        } else {
            console.log("No hay suficiente stock");
        }
    } else {
        console.log("El medicamento no existe");
    }
}

File: farmacia.cs
using System;
using System.Collections.Generic;

class Program
{
    static List<string> nombres = new List<string>();
    static List<int> cantidades = new List<int>();
    static List<double> precios = new List<double>();
    static double totalVentas = 0;

    static void Main()
    {
        int opcion;

        do
        {
            Console.WriteLine("\nFARMACIA");
            Console.WriteLine("1. Registrar medicamento");
            Console.WriteLine("2. Consultar inventario");
            Console.WriteLine("3. Vender medicamento");
            Console.WriteLine("4. Total vendido");
            Console.WriteLine("5. Salir");
            Console.Write("Opción: ");

            opcion = int.Parse(Console.ReadLine());

            if (opcion == 1) Registrar();
            else if (opcion == 2) Consultar();
            else if (opcion == 3) Vender();
            else if (opcion == 4) MostrarTotal();

        } while (opcion != 5);
    }

    static void Registrar()
    {
        Console.Write("Nombre: ");
        string nombre = Console.ReadLine();

        Console.Write("Cantidad: ");
        int cantidad = int.Parse(Console.ReadLine());

        Console.Write("Precio: ");
        double precio = double.Parse(Console.ReadLine());

        nombres.Add(nombre);
        cantidades.Add(cantidad);
        precios.Add(precio);

        Console.WriteLine("Medicamento registrado");
    }

    static void Consultar()
    {
        Console.WriteLine("\nINVENTARIO:");
        for (int i = 0; i < nombres.Count; i++)
        {
            Console.WriteLine(
                nombres[i] +
                " | Cantidad: " + cantidades[i] +
                " | Precio: " + precios[i]
            );
        }
    }

    static void Vender()
    {
        Console.Write("Medicamento a vender: ");
        string buscar = Console.ReadLine();

        for (int i = 0; i < nombres.Count; i++)
        {
            if (nombres[i] == buscar)
            {
                Console.Write("Cantidad a vender: ");
                int venta = int.Parse(Console.ReadLine());

                if (venta <= cantidades[i])
                {
                    cantidades[i] -= venta;
                    totalVentas += venta * precios[i];
                    Console.WriteLine("Venta realizada");
                }
                else
                {
                    Console.WriteLine("No hay suficiente stock");
                }
                return;
            }
        }

        Console.WriteLine("Medicamento no encontrado");
    }

    static void MostrarTotal()
    {
        Console.WriteLine("Total vendido: " + totalVentas);
    }
}
